.github/workflows/eden_rpi.yml

```diff
@@ -0,0 +1,140 @@
+          ROL_API_KEY: ${{ secrets.ROL_API_KEY }}
+          ROL_PROJECT: ${{ secrets.ROL_PROJECT }}
+          ROL_OVPN_CONF_BASE64: ${{ secrets.ROL_OVPN_CONF_BASE64 }}
+      - name: Get EVE
+        uses: actions/checkout@v2
+        with:
+          path: eve
+      - name: Setup packages
+        run: |
+          sudo apt update && sudo apt upgrade
+          sudo apt install -y curl golang net-tools qemu binfmt-support qemu-user-static qemu-utils qemu-system-x86 qemu-system-aarch64 openvpn jq
+      - name: Setup OVPN connection to RoL RPi4 demo
+        id: vpn
+        run: |
+          echo "$ROL_OVPN_CONF_BASE64" | base64 -d > ./config.ovpn
+          sudo openvpn --config ./config.ovpn --daemon
+          until ip -f inet addr show tun0; do sleep 5; ip a; done
+          echo ::set-output name=ip::$(ip -f inet addr show tun0 | sed -En -e 's/.*inet ([0-9.]+).*/\1/p')
+        env:
+          ROL_OVPN_CONF_BASE64: ${{ secrets.ROL_OVPN_CONF_BASE64 }}
+      - name: Prepare eden
+        run: |
+          if [ -f ${{ github.workspace }}/eve/tests/eden/eden-version ]; then
+            EDEN_VERSION=$(cat ${{ github.workspace }}/eve/tests/eden/eden-version)
+          else
+            EDEN_VERSION=lfedge/eden:30c8b3f
+          fi
+          docker run -v $PWD:/out $EDEN_VERSION cp -a /eden/. /out/
+          sudo chown -R $(whoami) .
+      - name: Setup eden
+        run: |
+          ./eden config add default --devmodel=general --arch=arm64
+          ./eden config set default --key eve.tag --value="snapshot"
+          ./eden config set default --key eve.hv --value ${{ matrix.hv }}
+          ./eden config set default --key adam.eve-ip --value ${{ steps.vpn.outputs.ip }}
+          ./eden config set default --key registry.ip --value ${{ steps.vpn.outputs.ip }}
+          ./eden config set default --key=eden.tests --value=${{ github.workspace }}/eve/tests/eden
+          ./eden setup -v debug --netboot=true
+          ./eden start
+      - name: Get Rack of Labs CLI
+        uses: actions/checkout@v2
+        with:
+          repository: insei/rolgo-cli
+          path: rolgo-cli
+      - name: Build Rack of Labs CLI
+        run: |
+          cd rolgo-cli && go build
+      - name: Rent RPi4
+        id: rol
+        run: |
+          ipxe_cfg_url="http://${{ steps.vpn.outputs.ip }}:8888/eserver/ipxe.efi.cfg"
+          rent_id=$(./rolgo-cli/rolgo-cli rental create -p "$ROL_PROJECT" -m raspberry --model pi_4_model_b_8gb -n GHAction-snapshot-${{ matrix.hv }} -i "$ipxe_cfg_url")
+          echo ::set-output name=id::$(echo $rent_id)
+        env:
+          ROL_API_URL: ${{ secrets.ROL_API_URL }}
+          ROL_API_KEY: ${{ secrets.ROL_API_KEY }}
+          ROL_PROJECT: ${{ secrets.ROL_PROJECT }}
+      - name: Onboard RPi4
+        run: |
+          while [[ "$(./rolgo-cli/rolgo-cli rental get -p "$ROL_PROJECT" -i ${{ steps.rol.outputs.id }} | jq -r .machineState)" != "Ready" ]]; do sleep 10; echo "Waiting for EVE to load"; done
```

Comment:
Yeah -- this place -- I REALLY would like Eden to import this code the same way @giggsoff made Eden talk natively to Packet.

Reply:
@rvs packet API is stable and can be used by all users, but RoL API can be used now for CI only. For this reason, I have not yet integrated it into EDEN. I would like to see how this would work in CI and later integrate into eden. Perhaps this option is preferable?

Reply:
I think we still need to do it via Eden. But we should start with making sure the API lives as part of LF Edge.

Reply:
@rvs
https://github.com/lf-edge/eden/pull/739
Then this option will suit us?